record.ts

```diff
@@ -104,18 +104,31 @@ export function omit<A, P extends keyof A>(
   return out as { [K in keyof A]: K extends P ? never : A[K] };
 }
 
-export function pick<R, K extends keyof R>(
-  ...props: [K, K, ...K[]]
-): (ta: R) => Pick<R, K> {
-  return (ta) => {
-    const output: Partial<Pick<R, K>> = {};
-
-    for (const k of props) {
-      output[k] = ta[k];
-    }
-
-    return output as Pick<R, K>;
-  };
+/**
+ * Picks specified `keys` from a `record`. Value-space implementation of the
+ * [`Pick`](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys)
+ * utility type.
+ *
+ * @example
+ * import { pipe } from "./fns.ts";
+ * import { pick } from "./record.ts";
+ *
+ * pipe({ a: 1, b: 2, c: 3 }, pick(["a", "b"]))
+ * // { a: 1, b: 2 }
+ *
+ * @category combinators
+ */
+export function pick<T, K extends keyof T>(
+  keys: readonly K[],
+): (record: T) => Pick<T, K> {
+  return (record) =>
+    keys.reduce(
+      (prev, key) => ({
+        ...prev,
+        ...(key in record ? { [key]: record[key] } : {}),
+      }),
```

Comment:
This looks good but reduce operation here creates 2n-1 intermediate objects that are never used. While it's nice to remain pure, in cases like this I would prefer that we opt for space and performance to keep this operation light if it is used in a tight loop somewhere. Would you mind reverting to the for loop or using keys.forEach and mutably altering a single output object? ie.

```ts
export function pick<T, K extends keyof T>(
  keys: readonly K[],
): (record: T) => Pick<T, K> {
  return (record) =>
    keys.reduce(
      (prev, key) => {
        if (key in record) {
          prev[key] = record[key];
        }
        return prev;
      },
      {} as Pick<T, K>,
    );
}

// Or 

export function pick<T, K extends keyof T>(
  keys: readonly K[],
): (record: T) => Pick<T, K> {
  return (record) => {
    const output: Pick<T, K> = {};
    for (const key of keys) {
      if (key in record) {
        output[key] = record[key];
      }
    }
    return output;
  }
}
```

Reply:
Sure. I vote for the vanilia `for` loop if you don’t mind.